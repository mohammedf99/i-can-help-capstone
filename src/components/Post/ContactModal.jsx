import React, { useState } from "react";
import { useTranslation } from "next-i18next";
import { Row, Col } from "antd";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faComments,
  faEnvelope,
  faPhoneAlt,
  faMapMarkerAlt,
} from "@fortawesome/free-solid-svg-icons";
import { faFacebook } from "@fortawesome/free-brands-svg-icons";
import { ContactBtn, ConModal } from "./Post.styled";

const ContactModal = ({ button, user }) => {
  const [isModalVisible, setIsModalVisible] = useState(false);
  const { t } = useTranslation("common");

  const showModal = () => {
    setIsModalVisible(true);
  };

  const handleOk = () => {
    setIsModalVisible(false);
  };

  const handleCancel = () => {
    setIsModalVisible(false);
  };

  return (
    <>
      {button ? (
        button({ onClick: () => showModal() })
      ) : (
        <ContactBtn type="primary" onClick={showModal}>
          <FontAwesomeIcon icon={faComments} className="msg-icon" /> {t("contact")}
        </ContactBtn>
      )}
      <ConModal
        title={`${user?.name} ${t("userContactInfo")}`}
        visible={isModalVisible}
        onOk={handleOk}
        onCancel={handleCancel}
        footer={null}
      >
        <Row gutter={[8, 8]}>
          {user?.contact?.email && (
            <Col span={24} className="col-style">
              <span className="icon">
                <FontAwesomeIcon icon={faEnvelope} />
              </span>
              {user?.contact?.email}
            </Col>
          )}
          {user?.contact?.facebook && (
            <Col span={24} className="col-style">
              <span className="icon">
                <FontAwesomeIcon icon={faFacebook} />
              </span>
              {user?.contact?.facebook}
            </Col>
          )}
          {user?.contact?.phone && (
            <Col span={24} className="col-style">
              <span className="icon">
                <FontAwesomeIcon icon={faPhoneAlt} />
              </span>
              {user?.contact?.phone}
            </Col>
          )}
          {user?.location && (
            <Col span={24} className="col-style">
              <span className="icon">
                <FontAwesomeIcon icon={faMapMarkerAlt} />
              </span>
              {user?.location}
            </Col>
          )}
        </Row>
      </ConModal>
    </>
  );
};

export default ContactModal;
