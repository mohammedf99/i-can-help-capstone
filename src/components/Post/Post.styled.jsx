import styled from "styled-components";
import { Tag, Button, Modal } from "antd";

export const PostDiv = styled.div`
  box-sizing: border-box;
  width: 99%;
  height: auto;
  padding: 2rem 3rem;
  display: flex;
  flex-direction: column;
  font-family: sans-serif;
  border: 1.5px solid #4e4e9f;
  border-radius: 5px;
  background-color: #fafaff;
  margin-bottom: 1rem;

  &:hover {
    border: 0;
    box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
    margin-bottom: 1.25rem;
  }

  .post-upper-section {
    display: flex;
    flex-direction: row;
  }

  .upper-left {
    display: flex;
    flex-direction: column;
    width: 60%;
  }

  .user-type {
    font-size: 18px;
    color: #4e4e9f;
    font-weight: 600;
  }

  .user-full-name {
    font-size: 22px;
    color: #1c1259;
    text-transform: uppercase;
    font-weight: 700;
  }

  .post-desc {
    font-size: 18px;
    color: #4e4e9f;
  }

  .upper-right {
    width: 40%;
    display: flex;
    align-items: center;
    justify-content: flex-end;
  }

  .post-lower-section {
    width: 100%;
    margin-top: 1rem;
    display: flex;
    flex-direction: row;
    align-items: center;
  }

  .lower-left {
    width: 60%;
  }

  .lower-right {
    width: 40%;
    display: flex;
    justify-content: flex-end;
  }

  @media only screen and (max-width: 768px) {
    padding: 1rem 2rem;
    
    .post-upper-section {
      flex-direction: column;
    }

    .upper-right,
    .upper-left {
      width: 100%;
    }

    .user-full-name {
      font-size: 18px;
    }

    .user-type {
      font-size: 16px;
    }

    .upper-right {
      justify-content: start;
    }
    .lower-left {
      width: 40%;
    }
    
    .lower-right {
      width: 60%;
      justify-content: flex-end;
      align-items: center;
    }
  }
`;

export const LocationTag = styled.div`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  border: 1px solid #ee4266;
  color: #ee4266;
  width: 95%;
  height: 40px;
  padding: 0.5rem 1rem;
  border-radius: 25px;
  font-size: 1rem;

  p {
    margin: 0;
    padding: 0;
  }

  .address {
    padding-left: 100px;
  }

  @media only screen and (max-width: 1200px) {
    .address {
      padding-left: 40px;
    }
  }

  @media only screen and (max-width: 992px) {
    .address {
      padding-left: 20px;
      font-size: 0.85rem;
    }
  }

  @media only screen and (max-width: 768px) {
    width: 50%;
    margin: 0;
    font-size: 0.75rem;
    .address {
      padding-left: 30%;
    }
  }

  @media only screen and (max-width: 600px) {
    .address {
      padding-left: 50px;
    }
  }
`;

export const Tags = styled(Tag)`
  color: #ee4266;
  padding: 0.5rem 1rem;
  border-radius: 25px;
  border: 1px solid #ee4266;
  margin-right: 0.5rem;
  background-color: transparent;

  @media only screen and (max-width: 600px) {
    padding: 0.25rem 0.5rem;
  }
`;

export const PinButton = styled.button`
  display: flex;
  color: #1c1259;
  background-color: transparent;
  border: 1px solid #1c1259;
  border-radius: 25px;
  padding: 0.5rem 1rem;
  margin-right: 0.5rem;
  font-size: 1rem;
  width: 175px;

  &:hover {
    background-color: #1c1259;
    color: #fafaff;
    outline: none;
    transition: all 0.5s 0s ease;
  }

  .pin-icon {
    margin-right: 50px;
    align-self: center;
  }

  @media only screen and (max-width: 1200px) {
    .pin-icon {
      margin-right: 30px;
    }
  }

  @media only screen and (max-width: 992px) {
    .pin-icon {
      margin-right: 10px;
    }
  }

  @media only screen and (max-width: 768px) {
    width: 120px;
    height: 35px;
    align-items: center;

    .pin-icon {
      margin-right: 25px;
    }
  }

  @media only screen and (max-width: 600px) {
    width: 100px;
    height: 30px;
    font-size: 0.75rem;
  }

`;

export const ContactBtn = styled(Button)`
  background-color: transparent;
  border: 1px solid #1c1259;
  width: 175px;
  height: 45px;
  color: #1c1259;
  border-radius: 25px;
  padding: 0.5rem 1rem;
  font-size: 1rem;

  &:hover {
    background-color: #1c1259;
    color: #fafaff;
    outline: none;
    transition: all 0.5s 0s ease;
  }

  .msg-icon {
    margin: 0px 30px 0px -30px;
  }

  &:focus {
    background-color: #cacaea;
    border: none;
    outline: none;
  }


  @media only screen and (max-width: 1200px) {
    .msg-icon {
      margin: 0px 15px 0px -15px;
    }
  }

  @media only screen and (max-width: 992px) {
    .msg-icon {
      margin: 0px 10px 0px -10px;
    }
  }
  
  @media only screen and (max-width: 768px) {
    width: 120px;
    height: 35px;
    display: flex;
    align-items: center;

    .msg-icon {
      margin: 0px 10px 0px -5px;
    }

    @media only screen and (max-width: 600px) {
      width: 100px;
      height: 30px;
      font-size: 0.75rem;
    }
`;

export const ConModal = styled(Modal)`
  .col-style {
    width: fit-content;
    box-sizing: border-box;
    font-size: 1rem;
    border: 1px solid #1c1259;
    border-radius: 5px;
    padding: 0.5rem 1rem;
    color: #1c1259;
    transition: all 0.5s 0s ease-in-out;
  }

  .col-style:hover {
    color: #ee4266;
    border: 1px solid #ee4266;
    font-size: 1.25rem;
    transition: all 0.5s 0s ease-in-out;
  }

  .icon {
    margin: 0 1rem 0 1rem;
  }
`;
