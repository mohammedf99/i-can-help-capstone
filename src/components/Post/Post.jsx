import React, { useState, useEffect, useContext } from "react";
import { useRouter } from "next/router";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faMapMarkerAlt, faCheck } from "@fortawesome/free-solid-svg-icons";
import {
  PostDiv,
  LocationTag,
  Tags,
  PinButton,
  PinnedButton,
} from "./Post.styled";
import ContactModal from "./ContactModal";
import {
  pinPost,
  unPinPost,
  usersRef,
} from "../../Utilities/FirebaseUtilities";
import AuthContext from "../../Utilities/Contexts/AuthContext";
import DataContext from "../../Utilities/Contexts/DataContext";

const Post = ({ data }) => {
  const { currentUser } = useContext(AuthContext);
  const { userData } = useContext(DataContext);
  const [user, setUser] = useState(null);

  useEffect(() => {
    if (data && data.userId) {
      usersRef
        .doc(data?.userId)
        .onSnapshot((snapshot) => setUser(snapshot.data()));
    }
  }, [data]);

  const router = useRouter();
  return (
    <PostDiv onClick={() => router.push(`/postDetails/${data?.id}`)}>
      <div className="post-upper-section">
        <div className="upper-left">
          <span className="user-type">
            {data?.type === 1 ? "Employeer" : "Jobseeker"}
          </span>
          <span className="user-full-name">{user?.name}</span>
          <p className="post-desc">{data?.title}</p>
        </div>
        <div className="upper-right">
          <LocationTag className="post-location-tag">
            <FontAwesomeIcon
              icon={faMapMarkerAlt}
              className="pink-icon-color"
            />
            <p className="address">{data?.location}</p>
          </LocationTag>
        </div>
      </div>
      <div className="post-lower-section">
        <div className="lower-left">
          <Tags>{data?.employment}</Tags>
          <Tags>{data?.jobType}</Tags>
        </div>
        <div className="lower-right">
          {userData?.pinnedPosts?.includes(data?.id) ? (
            <PinnedButton
              onClick={(e) => {
                e.stopPropagation();
                unPinPost(currentUser?.uid, data?.id);
              }}
            >
              <FontAwesomeIcon icon={faCheck} className="pin-icon" />
              Pinned
            </PinnedButton>
          ) : (
            <PinButton
              onClick={(e) => {
                e.stopPropagation();
                pinPost(currentUser?.uid, data?.id);
              }}
            >
              <FontAwesomeIcon icon={faCheck} className="pin-icon" />
              Pin
            </PinButton>
          )}

          <div onClick={(e) => e.stopPropagation()}>
            <ContactModal user={user} />
          </div>
        </div>
      </div>
    </PostDiv>
  );
};

export default Post;
