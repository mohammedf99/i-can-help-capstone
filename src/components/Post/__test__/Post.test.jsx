import * as React from "react";
import * as renderer from "react-test-renderer";
import Post from "../Post";

it("renders Post component correctly", () => {
  const tree = renderer.create(<Post />).toJSON();
  expect(tree).toMatchSnapshot();
});