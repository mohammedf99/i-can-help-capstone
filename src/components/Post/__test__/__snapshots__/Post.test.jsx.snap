// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Post component correctly 1`] = `
<div
  className="Poststyled__PostDiv-sc-r6hdig-0 hlWAfY"
  onClick={[Function]}
>
  <div
    className="post-upper-section"
  >
    <div
      className="upper-left"
    >
      <span
        className="user-type"
      >
        Jobseeker
      </span>
      <span
        className="user-type"
      >
        $
        /
      </span>
      <span
        className="user-full-name"
      />
      <p
        className="post-desc"
      />
    </div>
    <div
      className="upper-right"
    >
      <div
        className="Poststyled__LocationTag-sc-r6hdig-1 eQGOkH post-location-tag"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-map-marker-alt fa-w-12 pink-icon-color"
          data-icon="map-marker-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <p
          className="address"
        />
      </div>
    </div>
  </div>
  <div
    className="post-lower-section"
  >
    <div
      className="lower-left"
    >
      <span
        className="ant-tag Poststyled__Tags-sc-r6hdig-2 cKVDAJ"
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
      />
      <span
        className="ant-tag Poststyled__Tags-sc-r6hdig-2 cKVDAJ"
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
      />
    </div>
    <div
      className="lower-right"
    >
      <button
        className="Poststyled__PinButton-sc-r6hdig-3 kmYTTo"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 pin-icon"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        pin
      </button>
      <div
        onClick={[Function]}
      >
        <button
          className="ant-btn ant-btn-primary Poststyled__ContactBtn-sc-r6hdig-5 hQJVCu"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-comments fa-w-18 msg-icon"
            data-icon="comments"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 192c0-88.4-93.1-160-208-160S0 103.6 0 192c0 34.3 14.1 65.9 38 92-13.4 30.2-35.5 54.2-35.8 54.5-2.2 2.3-2.8 5.7-1.5 8.7S4.8 352 8 352c36.6 0 66.9-12.3 88.7-25 32.2 15.7 70.3 25 111.3 25 114.9 0 208-71.6 208-160zm122 220c23.9-26 38-57.7 38-92 0-66.9-53.5-124.2-129.3-148.1.9 6.6 1.3 13.3 1.3 20.1 0 105.9-107.7 192-240 192-10.8 0-21.3-.8-31.7-1.9C207.8 439.6 281.8 480 368 480c41 0 79.1-9.2 111.3-25 21.8 12.7 52.1 25 88.7 25 3.2 0 6.1-1.9 7.3-4.8 1.3-2.9.7-6.3-1.5-8.7-.3-.3-22.4-24.2-35.8-54.5z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span>
             contact
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
